core/shared/src/main/scala/org/http4s/websocket/WebSocketContext.scala

```diff
@@ -22,20 +22,45 @@ import cats.syntax.all._
 import cats.~>
 import org.typelevel.ci.CIStringSyntax
 
+import scala.concurrent.duration.FiniteDuration
+
 final case class WebSocketContext[F[_]](
```

Comment:
Damn, this is a case class. To properly evolve it we must follow all the steps here.

https://docs.scala-lang.org/overviews/core/binary-compatibility-for-library-authors.html#changing-a-case-class-definition-in-a-backwards-compatible-manner

Reply:
I tried and failed :/

Reply:
Which one of the steps in the guide did you get stuck on? That is the official Scala guide, so if there are problems with it we should open an issue there. There's also a WIP PR to improve it. https://github.com/scala/docs.scala-lang/pull/2788

Reply:
With defaults in `WebSocketContext#copy` it compiles and binchecks for Scala 2, fails to compile for Scala 3 (2 methods with default parameters)

Without defaults in `WebSocketContext#copy`it fails mima both for Scala 2 and 3.

Reply:
I tried to move the stuff to `WebSocketSeparatePipe` and `WebSocketCombinedPipe`, but with both being `final case class`, it is virtually impossible not sneak an extra parameter (I failed on `unapply`).